tests/tests_integration/test_api/test_spaces.py

```diff
@@ -0,0 +1,67 @@
+from __future__ import annotations
+
+import pytest
+
+from cognite.client import CogniteClient
+from cognite.client.data_classes import Space, SpaceList
+
+
+@pytest.fixture(scope="function")
+def cdf_spaces(cognite_client):
+    spaces = cognite_client.data_modeling.spaces.list(limit=-1)
+    assert len(spaces) > 0, "Please create at least one space in CDF."
+    yield spaces
+
+
+def _dump(list_: SpaceList | Space) -> list[dict]:
+    if isinstance(list_, Space):
+        return [list_.dump()]
+    return sorted((s.dump() for s in list_), key=lambda s: s["space"])
```

Comment:
You can optionally use the objects' existing dump methods and check equality with `unittest.TestCase().assertCountEqual(a, b)` which ignores ordering.

Reply:
This method also accounts for a Space equal to SpaceList with one element.